src/coreclr/jit/gcinfo.cpp

```diff
@@ -254,6 +254,15 @@ GCInfo::WriteBarrierForm GCInfo::gcIsWriteBarrierCandidate(GenTreeStoreInd* stor
 
     if (store->Data()->IsIconHandle(GTF_ICON_OBJ_HDL))
     {
+#ifndef TARGET_XARCH
+        const ssize_t handle = store->Data()->AsIntCon()->IconValue();
+        if (!compiler->info.compCompHnd->isObjectImmutable(reinterpret_cast<void*>(handle)))
+        {
+            // See https://github.com/dotnet/runtime/pull/76135#issuecomment-1257258310
```

Comment:
It would be nice with an inline comment condensing the discussion of this that explains why this is needed, perhaps with a reference to the memory model spec.

Reply:
Added more comments, the memory model doc is not published yet